benchmarking/compare.sh

```diff
@@ -0,0 +1,100 @@
+#!/bin/bash
+
+workdir=$(cd $(dirname $0); pwd)/work
+
+usage="Usage: $0 [-h] <hash1> <hash2>"
+
+while [ $# -gt 0 ]; do
+    case $1 in
+        -h|--help)
+            echo "${usage}"
+            exit 1
+            ;;
+        -*)
+            echo "Error: invalid option '$1'"
+            exit 1
+            ;;
+        *)
+            break
+    esac
+    shift; shift
+done
+
+if [ $# != 2 ]; then
+    echo ${usage}
+    exit 1
+fi
+
+benchmark1=$workdir/$1
+benchmark2=$workdir/$2
+
+lib1=$workdir/lib-$(basename $benchmark1)
+lib2=$workdir/lib-$(basename $benchmark2)
+
+if [ ! -x $benchmark1 ]; then
+    echo "Not executable: $benchmark1"
+    exit 1;
+fi
+
+if [ ! -x $benchmark2 ]; then
+    echo "Not executable: $benchmark2"
+    exit 1;
+fi
+
+if [ ! -d $lib1 ]; then
+    echo "library directory not found: $lib1"
+    exit 1;
+fi
+
+if [ ! -d $lib2 ]; then
+    echo "library directory not found: $lib2"
+    exit 1;
+fi
+
+TMPDIR=$(mktemp -d)
+trap "rm -rf ${TMPDIR}" EXIT
+
+tests=`LD_LIBRARY_PATH=$lib1 $benchmark1 --benchmark_list_tests`
+
+echo "Running benchmark $benchmark1 three times and saving the last" >&2
+for ((j=0; j < 3; j++)); do
+    LD_LIBRARY_PATH=$lib1 $benchmark1 --benchmark_format=csv \
+        |csvcut -c 'name,real_time' > $TMPDIR/bm1.csv
+done
+
+echo "Running benchmark $benchmark2 three times and saving the last" >&2
+for ((i=0; i < 3; i++)); do
+    LD_LIBRARY_PATH=$lib2 $benchmark2 --benchmark_format=csv \
+        |csvcut -c 'name,real_time' > $TMPDIR/bm2.csv
+done
+
+echo "set term pngcairo color"
+
+echo -n "set xtics ("
```

Comment:
Apparently,
```
echo -n "text"
```

prints `-n text` on Mac Os. Can this be done without the option?

Reply:
But not in the command line, only in the script?? Maybe I am doing something wrong...

Reply:
The shell is wrong.  It should be `/bin/bash`.  I'll fix it.